### PR TITLE
RHSTOR-5962: Disable disaster recovery for discovered applications

### PR DESCRIPTION
**Pull request by GowthamShanmugam**
Issue: https://issues.redhat.com/browse/RHSTOR-5962

ToDo: https://github.com/red-hat-storage/odf-console/issues/1449

### CODE REVIEW

File: packages/mco/components/modals/remove-disaster-recovery/remove-disaster-recovery.tsx [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+      },
+      data: patch,
+    })
+      .then(() =>
+        // Delete DRPC only after updating the annotation
+        k8sDelete({
+          resource: application,
+          model: DRPlacementControlModel,
+          json: null,
+          requestInit: null,
```
Comment (SanjalKatiyar) [inline packages/mco/components/modals/remove-disaster-recovery/remove-disaster-recovery.tsx:61]: do UI really need to wait for patch to happen and then delete ??
can't we send both requests simultaneously ??

File: packages/mco/components/modals/remove-disaster-recovery/remove-disaster-recovery.tsx [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        setInProgress(false);
+      });
+  };
+
+  return (
+    <Modal
+      variant={ModalVariant.small}
+      header={
+        <ModalHeader>
+          <ExclamationTriangleIcon color="#f0ab00" className="icon--spacer" />
```
Comment (SanjalKatiyar) [inline packages/mco/components/modals/remove-disaster-recovery/remove-disaster-recovery.tsx:82]: plz import color/icon from PF, if/whenever needed...

File: packages/mco/components/modals/remove-disaster-recovery/remove-disaster-recovery.tsx [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+          data-test="cancel-action"
+        >
+          {t('Cancel')}
+        </Button>
+        <Button
+          key="remove"
+          variant={ButtonVariant.danger}
+          onClick={onRemove}
+          data-test="remove-action"
+          isDisabled={isInprogress}
```
Comment (SanjalKatiyar) [inline packages/mco/components/modals/remove-disaster-recovery/remove-disaster-recovery.tsx:118]: ??
```suggestion
          isDisabled={isInprogress || error}
```

File: packages/mco/components/modals/remove-disaster-recovery/remove-disaster-recovery.tsx (hunk 1 above)
Comment (GowthamShanmugam) [reply packages/mco/components/modals/remove-disaster-recovery/remove-disaster-recovery.tsx:61]: It prevents the PVC deletion: https://access.redhat.com/articles/7057753

File: packages/mco/components/modals/remove-disaster-recovery/remove-disaster-recovery.tsx (hunk 3 above)
Comment (GowthamShanmugam) [reply packages/mco/components/modals/remove-disaster-recovery/remove-disaster-recovery.tsx:118]: they can't retry right?

**Comment by GowthamShanmugam** [comment]
image

File: packages/mco/components/modals/remove-disaster-recovery/remove-disaster-recovery.tsx [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+      json: null,
+      requestInit: null,
+    })
+  );
+
+  promises.push(
+    k8sDelete({
+      resource: {
+        metadata: {
+          name: name,
```
Comment (SanjalKatiyar) [inline packages/mco/components/modals/remove-disaster-recovery/remove-disaster-recovery.tsx:71]: same here... some common utils used every place...

Comment (SanjalKatiyar) [reply packages/mco/components/modals/remove-disaster-recovery/remove-disaster-recovery.tsx:71]: but okay to skip for now... will not block the PR on this...

File: packages/mco/components/modals/remove-disaster-recovery/remove-disaster-recovery.tsx [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+    model: DRPlacementControlModel,
+    resource: {
+      metadata: {
+        name: getName(application),
+        namespace: getNamespace(application),
+      },
+    },
+    data: patch,
+  });
+};
```
[the remark this answers is not in the document]
Comment (SanjalKatiyar) [reply packages/mco/components/modals/remove-disaster-recovery/remove-disaster-recovery.tsx:49]: but okay to skip for now... will not block the PR on this...

File: packages/mco/components/modals/remove-disaster-recovery/remove-disaster-recovery.tsx [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+    setInProgress(true);
+
+    pvcAnnotationPatchPromise(application)
+      .then(() => {
+        Promise.all(deleteApplicationResourcePromise(application))
+          .then(() => {
+            closeModal();
+          })
+          .catch((err) => {
+            setError(err);
```
Comment (SanjalKatiyar) [inline packages/mco/components/modals/remove-disaster-recovery/remove-disaster-recovery.tsx:103]: same here...

File: packages/mco/components/modals/remove-disaster-recovery/remove-disaster-recovery.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (Chandan141996) [reply packages/mco/components/modals/remove-disaster-recovery/remove-disaster-recovery.tsx:50]: There's a spelling mistake. It should be 'preventing', please fix it later.
@SanjalKatiyar

**Pull request merged**